extensions/tokenizers/src/main/java/ai/djl/huggingface/tokenizers/HuggingFaceTokenizer.java

```diff
@@ -535,4 +480,152 @@ static PaddingStrategy fromValue(String value) {
+         * @return this builder
+         */
+        public Builder optTokenizerName(String tokenizerName) {
+            options.put("tokenizer", tokenizerName);
+            return this;
+        }
+
+        /**
+         * Sets the file path of the tokenizer.
+         *
+         * @param tokenizerPath the path of the tokenizer
+         * @return this builder
+         */
+        public Builder optTokenizerPath(Path tokenizerPath) {
+            this.tokenizerPath = tokenizerPath;
+            return this;
+        }
+
+        /**
+         * Sets if add special tokens.
+         *
+         * @param addSpecialTokens true to add special tokens
+         * @return this builder
+         */
+        public Builder optAddSpecialTokens(boolean addSpecialTokens) {
+            options.put("addSpecialTokens", String.valueOf(addSpecialTokens));
+            return this;
+        }
+
+        /**
+         * Enables or Disables default truncation behavior for the tokenizer.
+         *
+         * @param enabled whether to enable default truncation behavior
+         * @return this builder
+         */
+        public Builder optTruncation(boolean enabled) {
+            options.put("truncation", String.valueOf(enabled));
+            return this;
+        }
+
+        /**
+         * Enables truncation to only truncate the first item.
+         *
+         * @return this builder
+         */
+        public Builder optTruncateFirstOnly() {
+            options.put("truncation", TruncationStrategy.ONLY_FIRST.name());
+            return this;
+        }
+
+        /**
+         * Enables truncation to only truncate the second item.
+         *
+         * @return this builder
+         */
+        public Builder optTruncateSecondOnly() {
+            options.put("truncation", TruncationStrategy.ONLY_SECOND.name());
+            return this;
+        }
+
+        /**
+         * Enables or Disables default padding behavior for the tokenizer.
+         *
+         * @param enabled whether to enable default padding behavior
+         * @return this builder
+         */
+        public Builder optPadding(boolean enabled) {
+            options.put("padding", String.valueOf(enabled));
+            return this;
+        }
+
+        /**
+         * Enables padding to pad sequences to previously specified maxLength, or modelMaxLength if
+         * not specified.
+         *
+         * @return this builder
+         */
+        public Builder optPadToMaxLength() {
+            options.put("padding", PaddingStrategy.MAX_LENGTH.name());
+            return this;
+        }
+
+        /**
+         * Sets maxLength for padding and truncation.
+         *
+         * @param maxLength the length to truncate and/or pad sequences to
+         * @return this builder
+         */
+        public Builder optMaxLength(int maxLength) {
+            options.put("maxLength", String.valueOf(maxLength));
+            return this;
+        }
```

Comment:
I think this is unlikely, but a user could pass in a length of 0 here by accident - should we treat 0 similar to how we treat -1 currently (unset, the default returned from our JNI)?